src/WebJobs.Script/Rpc/ILanguageWorkerChannel.cs

```diff
@@ -5,12 +5,18 @@
 
 namespace Microsoft.Azure.WebJobs.Script.Rpc
 {
-    internal delegate ILanguageWorkerChannel CreateChannel(WorkerConfig conf, IObservable<FunctionRegistrationContext> registrations, int attemptCount);
+    public delegate ILanguageWorkerChannel CreateChannel(string language, IObservable<FunctionRegistrationContext> registrations, int attemptCount);
 
-    internal interface ILanguageWorkerChannel : IDisposable
+    public interface ILanguageWorkerChannel : IDisposable
     {
         string Id { get; }
 
         WorkerConfig Config { get; }
+
+        void RegisterFunctions(IObservable<FunctionRegistrationContext> functionRegistrations);
```

Comment:
Is this a one time init for the channel? Can the method just be called Initialize?

Reply:
This is can be called at the time of LanguageWorkerChannel creation or when at a later point in time after creation. Register makes more sense as this method setups event registrations for functions discovered.